pydantic/schema.py

```diff
@@ -946,6 +952,43 @@ def constraint_func(**kwargs: Any) -> Type[Any]:
 
     ans = go(annotation)
 
+    check_unused_constraints(constraints, used_constraints, field_name)
+
+    return ans
+
+
+def check_unused_validation_assignment_constraints(
```

Comment:
this function name is too long.

Reply:
`check_unused_assignment_constraints` is the best alternative I can come up with at the moment.  Please suggest another name if you do not like this one.

Reply:
removed the function.